src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -1699,15 +1699,25 @@ otError RadioSpinel<InterfaceType, ProcessContextType>::WaitResponse(void)
 template <typename InterfaceType, typename ProcessContextType>
 spinel_tid_t RadioSpinel<InterfaceType, ProcessContextType>::GetNextTid(void)
 {
-    spinel_tid_t tid = 0;
+    spinel_tid_t tid = mCmdNextTid;
 
-    if (((1 << mCmdNextTid) & mCmdTidsInUse) == 0)
+    while (((1 << tid) & mCmdTidsInUse) != 0)
```

Comment:
`mCmdTidsInUse` is `uint16_t`, so the valid TID range should be [1~15].

Does the issue mean that we should increase the range of TID to [1~31] ?

@abtink @bukepo

Reply:
TID is 4 bit value in spinel protocol. So its range [0-15]
https://github.com/openthread/openthread/blob/119df62cbddce9d3e10b0a56d7fbf9d53c7fa218/src/lib/spinel/spinel.h#L59-L85

> Does the issue mean that we should increase the range of TID to [1~31] ?

No. We don't need do. 

`RadioSpinel` seems to only use at most two TIDs at a time (one for radio tx frame, one for a command for which it is waiting for response). The issue here was that `GetNextTid()` only checked one value `mCmdNxtTid` and not the rest of TID values.  More details on this in #7513.

Reply:
I got what you mean. Thanks for clarification.